File: eventx-backend/src/main/java/com/eventx/service/EventService.java
package com.eventx.service;

import com.eventx.dto.request.CreateEventRequest;
import com.eventx.dto.request.UpdateEventRequest;
import com.eventx.dto.response.EventResponse;
import com.eventx.dto.response.EventSummaryResponse;
import com.eventx.dto.response.SeatResponse;
import com.eventx.entity.enums.EventCategory;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.time.LocalDate;
import java.util.List;

public interface EventService {
    EventResponse createEvent(CreateEventRequest request, Long organizerId);
    EventResponse updateEvent(Long id, UpdateEventRequest request, Long actorId);
    void deleteEvent(Long id, Long actorId);
    EventResponse publishEvent(Long id, Long actorId);
    EventResponse getEvent(Long id);
    Page<EventSummaryResponse> getAllEvents(Pageable pageable);
    Page<EventSummaryResponse> getEventsByOrganizer(Long organizerId, Pageable pageable);
    Page<EventSummaryResponse> getPublishedEvents(Pageable pageable);
    Page<EventSummaryResponse> searchEvents(String query, EventCategory category, String city, LocalDate dateFrom, LocalDate dateTo, Pageable pageable);
    Page<EventSummaryResponse> getUpcomingEvents(Pageable pageable);
    List<SeatResponse> getAvailableSeats(Long eventId);
}

File: eventx-frontend/src/services/eventService.js
import api from './api';

const eventService = {
  getEvents: (params) => api.get('/events', { params }),
  getEvent: (id) => api.get(`/events/${id}`),
  searchEvents: (params) => api.get('/events/search', { params }),
  getUpcomingEvents: (params) => api.get('/events/upcoming', { params }),
  getSeats: (eventId) => api.get(`/events/${eventId}/seats`),
  // Admin
  createEvent: (data) => api.post('/admin/events', data),
  updateEvent: (id, data) => api.put(`/admin/events/${id}`, data),
  deleteEvent: (id) => api.delete(`/admin/events/${id}`),
  getAllEvents: (params) => api.get('/admin/events', { params }),
};

export default eventService;
